### PR TITLE
#165020203 Logout the user and blacklist user token

### PR DESCRIPTION
**Pull request by avpaul**
## What does this PR do?
Add logout/signout feature

## Description of Task to be completed?
The user should be able to drop out a token as a way of signing out.

## How should this be manually tested?
Postman

**For testing user logout**

0. Use git to clone the branch `git checkout -t origin ft-165020203-user-signout` and create the database by running this npm command `npm run migrate`
1. Create a user account and Log in  to generate a token
2. Sign out the user to blacklist the token(Copy the token returned when you log in and paste it in the Authorization tab in postman)

`localhost:3000/api/v1/auth/login` then `localhost:3000/api/v1/auth/logout/`
Use the blacklisted token to access a protected route
`/api/v1/protected-route`

**For signup errors**

0. Use git to clone the branch `git checkout -t origin ft-165020203-user-signout` and create the database by running this npm command `npm run migrate`
1. Create a user account `localhost:3000/api/v1/auth/signup`, Content-Type =application/x-www-form-urlencoded and pass email password and username
2. Try different combinations

## What are the relevant pivotal tracker stories?
[User should be able to sign out #165020203](https://www.pivotaltracker.com/n/projects/2321840)

### CODE REVIEW

**Comment by avpaul** [comment]
Travis is failing because of the logout test which requires token verification

**Review by aaronsekisambu — DISMISSED** [review]

**Comment by daud1** [comment]
Fix your build before this can be reviewed or merged.

**Comment by avpaul** [comment]
@daud1 I am waiting for @shemaeric for tokens so that I can fix it.

**Review by aaronsekisambu — DISMISSED** [review]

File: index.js
Comment (niyobobo) [inline index.js]: It's better if you don't expose this server information in the console

File: middlewares/userValidation.js
Comment (niyobobo) [inline middlewares/userValidation.js]: I think it should be better if you don't pass the server error to the controller. you should return the response to the user request with that error.

File: tests/base.test.js
Comment (niyobobo) [inline tests/base.test.js]: Better if you don't expose your token outside.

Comment (avpaul) [reply tests/base.test.js]: It was for test purpose

File: middlewares/userValidation.js
Comment (avpaul) [reply middlewares/userValidation.js]: As this error is not caused by user input I leave it to the general error handler middleware added in the index.js file to handle it.

**Review by fridolinho — COMMENTED** [review]
codes look good, you only have to fix conflicts.

**Review by shemaeric — APPROVED** [review]
After Testing and reviewing it looks good.

**Review by niyobobo — APPROVED** [review]
LGTM

**Review by fridolinho — APPROVED** [review]

**Review by aaronsekisambu — CHANGES_REQUESTED** [review]
@avpaul be more descriptive on how to test this PR manually for examples you can include what should be included in the body of the postman when testing this PR etc

**Review by daud1 — APPROVED** [review]
👍

**Review by aaronsekisambu — APPROVED** [review]

**Pull request merged**
